### ISSUE TITLE
CSharp client DateTimeOffset looses time zone

### ISSUE THREAD

**Issue by ili** [RSuter/NSwag#1007]
If `DateTimeOffset` is used as `DateTime` type time zone can be lost, because `"s"` format is used to covert value to string for HTTP parameter. It would be better to use `"O"` format, or in general to provide setting to define formats for `Date`, `DateTime` 'TimeSpan` and `Time` types

**Comment by RSuter** [comment]
I think there is a setting to change the format string

**Comment by RSuter** [comment]
ParameterDateTimeFormat

**Comment by ili** [comment]
@RSuter great! Thanks, I'v found it :) 

In any way it would be great also to make it configurable for other time types!

**Comment by RSuter** [comment]
You mean an additional ParameterDateFormat setting?

**Comment by ili** [comment]
Yes. We can specify C# types for Swagger Date, DateTime, Time & TimeSpan. From my side it looks good also to specify format for them separately as with ParameterDateTimeFormat for DateTime

**Comment by RSuter** [comment]
Currently, ParameterDateTimeFormat is used for Date and DateTime. TimeSpan is handled with ToString I think.

**Comment by IanKemp** [comment]
It's not just TZ info that is lost, `s` excludes milliseconds too. The default for `parameterDateTimeFormat` should be changed from `s` to `o`.
